src/test/java/benchmark/DFSelectionSetBenchmark.java

```diff
@@ -0,0 +1,117 @@
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.io.Resources.getResource;
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Mode.Throughput)
+@Warmup(iterations = 2)
+@Measurement(iterations = 2, timeUnit = TimeUnit.NANOSECONDS)
+public class DFSelectionSetBenchmark {
+
+    @State(Scope.Benchmark)
+    public static class MyState {
+
+        public ExecutableNormalizedField normalisedField;
+        public GraphQLOutputType outputFieldType;
+        GraphQLSchema schema;
+        Document document;
+
+        @Setup
+        public void setup() {
+            try {
+                String schemaString = readFromClasspath("large-schema-2.graphqls");
+                schema = SchemaGenerator.createdMockedSchema(schemaString);
+
+                String query = readFromClasspath("large-schema-2-query.graphql");
+                document = Parser.parse(query);
+
+                ExecutableNormalizedOperation executableNormalizedOperation = ExecutableNormalizedOperationFactory.createExecutableNormalizedOperation(schema, document, null, CoercedVariables.emptyVariables());
+
+                normalisedField = executableNormalizedOperation.getTopLevelFields().get(0);
+
+                outputFieldType = schema.getObjectType("Object42");
+
+            } catch (Exception e) {
+                System.out.println(e);
+                throw new RuntimeException(e);
+            }
+        }
+
+        private String readFromClasspath(String file) throws IOException {
+            URL url = getResource(file);
+            return Resources.toString(url, Charsets.UTF_8);
+        }
+    }
+
+    @Benchmark
+    @Warmup(iterations = 2)
+    @Measurement(iterations = 5, time = 10)
+    @Threads(1)
+    @Fork(3)
+    @BenchmarkMode(Mode.AverageTime)
+    @OutputTimeUnit(TimeUnit.MILLISECONDS)
+    public void benchMarkAvgTime(MyState myState, Blackhole blackhole) {
+        List<SelectedField> fields = getSelectedFields(myState);
+        blackhole.consume(fields);
+    }
+
+    @Benchmark
+    @Warmup(iterations = 2)
+    @Measurement(iterations = 5, time = 10)
+    @Threads(1)
+    @Fork(3)
+    @BenchmarkMode(Mode.Throughput)
+    @OutputTimeUnit(TimeUnit.MILLISECONDS)
+    public void benchMarkThroughput(MyState myState, Blackhole blackhole) {
+        List<SelectedField> fields = getSelectedFields(myState);
+        blackhole.consume(fields);
+    }
+
+    private List<SelectedField> getSelectedFields(MyState myState) {
+        DataFetchingFieldSelectionSet dataFetchingFieldSelectionSet = DataFetchingFieldSelectionSetImpl.newCollector(myState.schema, myState.outputFieldType, () -> myState.normalisedField);
+        return dataFetchingFieldSelectionSet.getFields("wontBeFound");
+    }
+
+    public static void mainX(String[] args) throws InterruptedException {
```

Comment:
Btw I found it's perfectly fine to simply main this `main`. In IntelliJ the run button on the class will run this, but the run buttons next to the `benchMarkAvgTime` and `benchMarkThroughput` functions still work fine.